### PR TITLE
Add ROADMAP to Atomic App

### PR DESCRIPTION
**Pull request by cdrage**
This would indicate the current direction of Atomic App as well as what is still needed to be implemented from the Nulecule spec.

A better indication of what is currently being worked on in comparison to https://trello.com/b/j1rEolFe/container-tools

### CODE REVIEW

**Comment by dustymabe** [comment]
I haven't done this before but this might actually be too detailed to the point where it will need a lot of maintenance. Also I think some of the stuff has the wrong priorities.. We need to get everyone who has a stake to weigh in on this:

@aweiteka ^^

File: ROADMAP.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#### Atomic App 1.0.0
+
+ - Docker compose provider [#265](https://github.com/projectatomic/atomicapp/issues/265)
+ - Ansible provider [#170](https://github.com/projectatomic/atomicapp/issues/170)
+ - Nspawn provider [#262](https://github.com/projectatomic/atomicapp/issues/262)
+ - Add a `USER` to Atomic App image [#252](https://github.com/projectatomic/atomicapp/issues/252)
+ - https/ssh/sftp support for artifacts
+ - Support running Kubernetes from an Openshift template [#131](https://github.com/projectatomic/atomicapp/issues/131)
+ - Persistent storage
```
Comment (vpavlin) [inline ROADMAP.md]: This should get much more priority

File: ROADMAP.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ - Refactoring OpenShift provider [#321](https://github.com/projectatomic/atomicapp/issues/321)
+
+#### Atomic App 1.0.0
+
+ - Docker compose provider [#265](https://github.com/projectatomic/atomicapp/issues/265)
+ - Ansible provider [#170](https://github.com/projectatomic/atomicapp/issues/170)
+ - Nspawn provider [#262](https://github.com/projectatomic/atomicapp/issues/262)
+ - Add a `USER` to Atomic App image [#252](https://github.com/projectatomic/atomicapp/issues/252)
+ - https/ssh/sftp support for artifacts
+ - Support running Kubernetes from an Openshift template [#131](https://github.com/projectatomic/atomicapp/issues/131)
```
Comment (vpavlin) [inline ROADMAP.md]: We'll need this sooner for demos

File: ROADMAP.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ - Implement stop for OpenShift provider [#135](https://github.com/projectatomic/atomicapp/issues/135)
+ - Refactoring OpenShift provider [#321](https://github.com/projectatomic/atomicapp/issues/321)
+
+#### Atomic App 1.0.0
+
+ - Docker compose provider [#265](https://github.com/projectatomic/atomicapp/issues/265)
+ - Ansible provider [#170](https://github.com/projectatomic/atomicapp/issues/170)
+ - Nspawn provider [#262](https://github.com/projectatomic/atomicapp/issues/262)
+ - Add a `USER` to Atomic App image [#252](https://github.com/projectatomic/atomicapp/issues/252)
+ - https/ssh/sftp support for artifacts
```
Comment (vpavlin) [inline ROADMAP.md]: I thought about it more and I don't think it's too important - I feel like packaging artifacts into a metadata image is much better practice than pulling them from somewhere

File: ROADMAP.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+#### Atomic App 0.4.0 (Late Dec?)
+ - Keep versioning info in one location [#155](https://github.com/projectatomic/atomicapp/issues/155)
+ - Implement stop for OpenShift provider [#135](https://github.com/projectatomic/atomicapp/issues/135)
+ - Refactoring OpenShift provider [#321](https://github.com/projectatomic/atomicapp/issues/321)
+
+#### Atomic App 1.0.0
+
+ - Docker compose provider [#265](https://github.com/projectatomic/atomicapp/issues/265)
+ - Ansible provider [#170](https://github.com/projectatomic/atomicapp/issues/170)
+ - Nspawn provider [#262](https://github.com/projectatomic/atomicapp/issues/262)
```
Comment (vpavlin) [inline ROADMAP.md]: Please also add https://github.com/projectatomic/atomicapp/issues/95 - multiple people asked for this recently

**Comment by vpavlin** [comment]
I've added comments to specific issues..generally, I'd say raise priority for anything Openshift. 

Anyway, I'd probably rather like to see a roadmap for Nulecule Library;)

File: ROADMAP.md (hunk 3 above)
Comment (charliedrage) [inline ROADMAP.md]: We will have to change the Nulecule spec though for this :)

**Comment by charliedrage** [comment]
Okay so I updated the roadmap from the feedback I've received.

pingarino @vpavlin arino

**Comment by aweiteka** [comment]
> I haven't done this before but this might actually be too detailed to the point where it will need a lot of maintenance. Also I think some of the stuff has the wrong priorities.. We need to get everyone who has a stake to weigh in on this:

I agree this is too specific. Dates are always misleading (dangerous?). I like the idea of a list of features and associated priority. This could be done as GH issue(s).

**Comment by charliedrage** [comment]
@aweiteka I agree with you that dates are dangerous, hence the disclaimer at the top. But I disagree that we should have the roadmap as github issue(s). Having a roadmap in a centralized location allows users to clearly see what we're working on and a centralized location on what needs to be done before 1.0.

Similar to:
https://github.com/coreos/rkt/blob/master/ROADMAP.md
https://github.com/docker/docker/blob/master/ROADMAP.md

**Comment by aweiteka** [comment]
I've had pretty good success using milestones flags on GH issues. It is how many people use Trello also. This makes it easy to quickly tag features to releases without a PR and all the discussion to accompany the PR. :)

Regarding the nulecule support statement that should probably go into another doc anyway. Top of the README? Programmatically? (return error if Nulecule ver is unsupported) `docs/requirements.md`?

**Comment by charliedrage** [comment]
@aweiteka Which Nulecule statement? Is it the one that relates to following the spec?

If so, yes, we should add that to docs/requirements.md :)

**Comment by aweiteka** [comment]
> Which Nulecule statement? Is it the one that relates to following the spec?

Yes. That's good info.

**Comment by charliedrage** [comment]
Added requirements.md, updated this PR. Any other comments?

**Comment by aweiteka** [comment]
@charliedrage Maybe I'm missing something but I'm not seeing the change.

**Comment by charliedrage** [comment]
Ah, I just added the "use API's for Docker and Kubernetes instead of CLI" as well as opened the updating the requirements.md PR yesterday.

**Comment by cdrage** [comment]
ping @dustymabe @aweiteka thoughts on the roadmap?

**Comment by aweiteka** [comment]
My comments still stand on making specific commitments. Pointing interested users to dynamic milestone links is much easier. Just a recommendation. I'm not a maintainer so go for it if you get the go-ahead. :)

**Comment by dustymabe** [comment]
I would remove "GA" or any references to the CDK from the Roadmap. It might be easier for us to just list items we want to do and say what their priority is (high/medium/low). Either way if you agree to maintain this file then I'm ok with it in the format you like.

**Comment by cdrage** [comment]
@dustymabe 
Yup. I'll maintain it. It gives us a good indication on what is currently on our plate for those who choose not to go on Trello.

roadmap updated again ^^

**Comment by dustymabe** [comment]
At one point you did have links to GH issues that corresponded to some of the entries. Do you recall why we took it out? Some of the items might be worth linking to a better description (like "refactor openshift") 

If you want  to merge as is.. it LGTM

**Comment by cdrage** [comment]
@dustymabe It'd be too much to maintain with all corresponding links you said from your first comment.

**Pull request merged**